core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -410,7 +410,7 @@ private[spark] class AppStatusListener(
 
       // Check if there are any pending stages that match this job; mark those as skipped.
       val it = liveStages.entrySet.iterator()
-      while (it.hasNext()) {
+      while (job.stageIds.nonEmpty && it.hasNext()) {
```

Comment:
Could you explain why we need this change?

Reply:
Ah, I suppose to refactor [this change](https://github.com/apache/spark/pull/26637/commits/44d174a139e01397ac134a22388f17bbaadca46d#diff-b43d5bd18e0f3b5f33d734a37389ed34R417) but it's wrong refactoring.
I'll revert this refactoring.